File: Code/LZ77-Compression/src/main/java/org/example/Util/Entry.java
package org.example.Util;

public class Entry {
    private int offset;
    private int length;
    private char nextSymbol;

    public Entry(){}
    public Entry(int offset, int length, char nextSymbol){
        this.offset = offset;
        this.length = length;
        this.nextSymbol = nextSymbol;
    }

    public int getOffset() {
        return offset;
    }

    public int getLength() {
        return length;
    }

    public char getNextSymbol() {
        return nextSymbol;
    }

    public void setOffset(int offset) {
        this.offset = offset;
    }

    public void setLength(int length) {
        this.length = length;
    }

    public void setNextSymbol(Character nextSymbol) {
        this.nextSymbol = nextSymbol;
    }
}

File: Code/javascript/LZ77-Kompression/src/Util/Entry.js
class Entry {

    constructor(offset, length, nextSymbol) {
      this.offset = offset;
      this.length = length;
      this.nextSymbol = nextSymbol;
    }

    getOffset() {
        return this.offset;
    }

    getLength() {
        return this.length;
    }

    getNextSymbol() {
        return this.nextSymbol;
    }

    setOffset(offset) {
        this.offset = offset;
    }

    setLength(length) {
        this.length = length;
    }

    setNextSymbol(nextSymbol) {
        this.nextSymbol = nextSymbol;
    }
}